github/resource_github_repository_environment.go

```diff
@@ -134,6 +140,8 @@ func resourceGithubRepositoryEnvironmentRead(d *schema.ResourceData, meta interf
 
 	d.Set("repository", repoName)
 	d.Set("environment", envName)
+	d.Set("wait_timer", nil)
```

Comment:
Do you mind explaining why setting the wait_timer to nil here solves the issue? It's not quite clear to me.

Reply:
When no wait_timer is removed then the API does not show the wait_timer setting at all when listing the environment.

So the following happens in the current code:

1. wait_timer is set in terraform

2. wait_timer is removed outside of terraform

3. We run a terraform apply and the code runs through lines 138-166 in resource_github_repository_environment.go (in the current main branch).

4. This code iterates over the protection rules BUT it never finds the wait_timer as it is just not shown at all in the API (if it were shown as "wait_timer = 0" it would find it, but its not there at all).

5. Therefore terraform never gets sent the current (ie 0) value of the wait_timer, so as far as terraform is concerned it has not changed.



Setting the wait_timer to "nil" means terraform now sees there is a diff at that point in time (note this is BEFORE terraform reports the diff to the user).

It then loops over the protection rules code.

If it does NOT find the wait_timer (ie it is removed/0 outside of terraform) then it already has it set to "nil" so it has a final diff to fix.


AND...


If it DOES find it in the protection rules (ie it has NOT been removed outside of terraform) is it compares that value with the value in the statefile (and we get a diff or not as normal, ie depending on whether the GitHUb value matches that of the statefile.